open_feature/open_feature_client.py

```diff
@@ -272,11 +344,30 @@ def _create_provider_evaluation(
         get_details_callable = {
             FlagType.BOOLEAN: self.provider.get_boolean_details,
             FlagType.NUMBER: self.provider.get_number_details,
+            FlagType.INTEGER: self.provider.get_number_details,
```

Comment:
One thing I'm not clear on (from reading the spec and comparing against other sdks, e.g. [the java sdk](https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/FeatureProvider.java)), is whether we should be providing all of these options backed up by a single method on the provider. 

It seems to me like we've settled in some middle ground where we're providing integer / float methods on the client but not encouraging providers to do the same. 

I would suggest that we either 'strictly' type on both sides or neither? I'm interested in other peoples' feedback here though? e.g. @justinabrahms @beeme1mr @tcarrio

Reply:
JS: https://github.com/open-feature/js-sdk/blob/main/src/no-op-provider.ts Using generic number

Java: https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/FeatureProvider.java Specific number types

Reply:
(Non-python engineer, so take that into account) I think I agree with @matthewelwell that it should be consistent. As far as if integer/number should be individually supporting, consider:

- support for fractional numbers could be important, especially <1. I could see feature flags that are proportions quite easily (e.g. 0.5)
- precision loss with high numbers (probably less important in feature flagging use-cases)

Reply:
I'm inclined to lean towards the Java approach here. It feels like the generic `number` approach suits JS better with the closer ties to JSON but in python terms 'number' doesn't really mean an awful lot. I'd suggest that we standardise to having 2 methods, one for int and one for float, both on the client and on the provider. 

That said, I'm open to being disagreed with here 😄

Reply:
Agreed I would always prefer stronger typing if possible. Python seems to be moving that way, better to future proof it.

Reply:
I'm also weak on the Python front but I'm all for stronger type safety. As long as a Python dev wouldn't be confused or put off by this, I would say go for it.